includes/Shortcode/Stories_Shortcode.php

```diff
@@ -123,15 +132,43 @@ private function prepare_story_attrs( array $attributes ): array {
 	 *
 	 * @since 1.5.0
 	 *
-	 * @param array<string,string|int> $attributes Array of arguments for Story Query.
-	 * @return array<string,string|int> Array of story arguments to pass to Story_Query.
+	 * @param array<string,int|string> $attributes Array of arguments for Story Query.
+	 * @return array<string,mixed> Array of story arguments to pass to Story_Query.
 	 */
 	private function prepare_story_args( array $attributes ): array {
-		return [
+		$args = [
 			// Show 100 stories at most to avoid 500 errors.
 			'posts_per_page' => min( (int) $attributes['number_of_stories'], 100 ), // phpcs:ignore WordPress.WP.PostsPerPage.posts_per_page_posts_per_page
 			'order'          => 'ASC' === $attributes['order'] ? 'ASC' : 'DESC',
 			'orderby'        => $attributes['orderby'],
 		];
+
+		$taxonomies           = get_object_taxonomies( Story_Post_Type::POST_TYPE_SLUG );
+		$should_add_tax_query = false;
+
+		foreach ( $taxonomies as $taxonomy ) {
+			if ( '' !== $attributes[ $taxonomy ] ) {
+				$should_add_tax_query = true;
+				break;
+			}
+		}
+
+		if ( $should_add_tax_query ) {
+			$i         = 0;
+			$tax_query = [
+				'relation' => 'OR',
+			];
+
+			foreach ( $taxonomies as $taxonomy ) {
+				$tax_query[ $i++ ] = [
```

Comment:
Why is the `$i++` increment needed here? Does `$tax_query[]` not work? Maybe because this is a mix of string and numeric array keys? If so, worst case we could potentially do something like this:

```php
$tax_query = [];

foreach ( $taxonomies as $taxonomy ) {
				$tax_query[] = [ ... ];
}

$tax_query['relation'] = 'OR';
```